### ISSUE TITLE
n breaks when bsdtar is used

### ISSUE THREAD

**Issue by sinapis** [tj/n#696]
# Bug Report

## Summary

when i try to install new version of node using n i get the following error:
```
n 17
  installing : node-v17.1.0
       mkdir : /usr/local/n/versions/node/17.1.0
       fetch : https://nodejs.org/dist/v17.1.0/node-v17.1.0-linux-x64.tar.gz
tar: Error opening archive: Failed to open '/dev/st0'
curl: (23) Failure writing output to destination
cp: cannot stat '/usr/local/n/versions/node/17.1.0/lib': No such file or directory
cp: cannot stat '/usr/local/n/versions/node/17.1.0/bin': No such file or directory
cp: cannot stat '/usr/local/n/versions/node/17.1.0/include': No such file or directory
cp: cannot stat '/usr/local/n/versions/node/17.1.0/share': No such file or directory
   installed : v14.18.1 (with npm 6.14.15)
```
my tar version is:
```
bsdtar 3.5.2 - libarchive 3.5.2 zlib/1.2.11 liblzma/5.2.5 bz2lib/1.0.8 liblz4/1.9.3 libzstd/1.5.0
```

I believe the problem is akin to what is described here https://unix.stackexchange.com/a/131053

## Steps to Reproduce
1. brew install libarchive
2. make sure /home/linuxbrew/.linuxbrew/bin appears before /bin/ in $PATH
3. n 17 (or some other non existent node version)

## Expected behavior
node 17 is installed successfully

## Actual behavior
error as above
also, n ls lists 17, even though it was installed and the folder /usr/local/n/versions/node/17.1.0/ is empty

## Other Information
if i switch to regular tar, all works fine

## Configuration Details

running in ubuntu WSL.

```bash
$ n --version
6.1.3

$ command -v node
/usr/local/bin/node

$ node -p process.platform
linux
```

**Comment by shadowspawn** [comment]
Thanks for detailed report, and reference.

`n` is not currently specifying `-f -` for tar, and I'll investigate using that.

**Comment by shadowspawn** [comment]
man page for freebsd tar has:

```
-f	file, --file file
	     Read the archive from or write the	archive	to the specified file.
	     The filename can be - for standard	input or standard output.  The
	     default varies by system; on FreeBSD, the default is /dev/sa0; on
	     Linux, the	default	is /dev/st0.
```

https://www.freebsd.org/cgi/man.cgi?query=bsdtar&apropos=0&sektion=1&manpath=FreeBSD+14.0-current&arch=default&format=html

**Comment by shadowspawn** [comment]
An old page, but answers with coverage of various tar versions: https://unix.stackexchange.com/questions/101561/what-are-the-differences-between-bsdtar-and-gnu-tar

**Comment by shadowspawn** [comment]
Fixed in `n` v8.0.1
